packages/puppeteer-core/src/cdp/ElementHandle.ts

```diff
@@ -201,4 +202,30 @@ export class CdpElementHandle<
       >;
     });
   }
+
+  override async renderedFonts(): Promise<PlatformFontUsage[]> {
+    try {
+      // In order to request a node and receive its node id, the document must be
+      // requested first
+      await this.client.send('DOM.getDocument', {depth: -1});
+      // In order to use a CSS-domain method, the CSS domain must be enabled
+      await this.client.send('CSS.enable');
+
+      const nodeInfo = await this.client.send('DOM.requestNode', {
+        objectId: this.id as string,
+      });
+      const nodeId = nodeInfo.nodeId;
+
+      const platformFonts = await this.client.send(
+        'CSS.getPlatformFontsForNode',
```

Comment:
I would not like to bring this method to Puppeteer as the method exists primarily for the DevTools frontend and it is very specific to the UI logic there. 

How about we document instead how to call this using a custom session created createCDPSession? Also, what is the use case for getting the font data via Puppeteer? If the use case is important, I think we could add this method as experimental but let us fix the [upstream method](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/inspector/inspector_css_agent.cc;l=1900;drc=277f4ab48eb85f7441f78aed191c31068ce89814) first so that it does not require DOM and CSS domains to be enabled and so that it accepts a backend node ID without the need to rely on the fragile DOM nodeIds (they reset on every call to DOM.getDocument so the method might fail if called concurrently).

Reply:
With https://github.com/puppeteer/puppeteer/pull/13328 the private property access will not be needed. Would you like to contribute to the fixes in Chromium?

Reply:
Sure. I am not familiar enough with the code base but this seems like a relatively minor issue, so maybe I can start with it

Reply:
here are the getting started instructions for building/testing https://www.chromium.org/developers/how-tos/get-the-code/ , it would require updating https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/public/devtools_protocol/browser_protocol.pdl;l=1;drc=93fc8bf077eb95148a50f71b9917ebe6bd5a02d3 to add optional objectId and backendNodeId params, then re-build to update the generated classes, and then in https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/inspector/inspector_css_agent.cc;l=1911;drc=b59cf80380080c28b981e2dcd491ef3efa9d8808 use the overload from here https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/inspector/inspector_dom_agent.cc;l=430;drc=1651676a30cd7abcd177975f7cd0e37bd945f663 Then the check if the agent is enabled can be removed and a test needs to be added verifying that it works with the new ids and also without the CSS domain enabled.